coreplugins/cesiumion/public/components/Fetcher.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { PureComponent } from "react";
+
+import AppContext from "./AppContext";
+import { fetchCancelable, getCookie } from "../utils";
+
+export class Fetcher extends PureComponent {
+	static defaultProps = {
+		url: "",
+		path: "",
+		method: "GET",
+		onBindRefresh: () => {},
+		onError: () => {},
+		onLoad: () => {}
+	};
+
+	state = {
+		isLoading: true,
+		isError: false
+	};
+
+	cancelableFetch = null;
+
+	fetch = () => {
+		const {
+			url,
+			path,
+			onError,
+			onLoad,
+			refresh,
+			children,
+			params,
+			...options
+		} = this.props;
+
+		let queryURL = `${url}/${path}`;
+		if (params !== undefined) {
```

Comment:
We might be able to clean this up with the URL object

https://developer.mozilla.org/en-US/docs/Web/API/URL